### PR TITLE
Fixes #59 & #69 Separate scripts and update docs

### LINKED ISSUE #59: Separating Documentation generation and Github Pages deployment
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

There is a single script for generating documents and publishing them to gh-pages branch. 
### Expected behaviour

Separating both these processes would enable many possibilities, including FTP deployment et al.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Separating generation and publishing of documents would enable many possibilities, including FTP deployment et al.
## Related Issue

Issue #59 & #69 
## How Has This Been Tested?

Tested on a local repository.

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by mariobehling** [comment]
@imujjwal96 Please actively ask your co-developers for reviews using the "@" functionality.

**Comment by imujjwal96** [comment]
I'm sorry. Will keep that in mind.

@sch00lb0y @pri22296 Please review.

**Comment by sch00lb0y** [comment]
could you give me your tested repo travis log @imujjwal96

**Comment by imujjwal96** [comment]
Here is the travis log. It fails because of that exception, but it calls the other script properly as the commands from that script are performed https://travis-ci.org/imujjwal96/prelimQuiz/jobs/238708224

File: generate.sh [hunk 1]
```diff
@@ -1,49 +1,51 @@
...
-git push origin gh-pages
-cd ../../
-rm -rf yaydocclone
-rm -rf yaydoctemp
-rm -rf yaydocvenv
-deactivate
-rm -- "$0"
+
+# Running the script to publish documentation
+source <(curl -s https://raw.githubusercontent.com/fossasia/yaydoc/master/publish_docs.sh)
```
Comment (pri22296) [inline generate.sh]: delete the yaydocvenv directory. also this script would not delete itself right?

Comment (imujjwal96) [inline generate.sh]: Oh! Didn't realise. Thanks for noticing.

Which script?

Comment (pri22296) [inline generate.sh]: this one. generate.sh. earlier it had a rm -- "$0" at the end that meant it deleted itself after use. now that line ha been  moved to publish_docs.sh so generate.sh would not get deleted right?

Comment (pri22296) [inline generate.sh]: this one. generate.sh. earlier it had a rm -- "$0" at the end that meant it deleted itself after use. now that line ha been moved to publish_docs.sh so generate.sh would not get deleted right?

Comment (imujjwal96) [inline generate.sh]: it will delete generate.sh

Comment (pri22296) [inline generate.sh]: sorry i'm not very well  versed with shell scripts. still learning.

Comment (imujjwal96) [inline generate.sh]: It is used to execute a script from URL

Comment (pri22296) [inline generate.sh]: oh. then generate.sh would be deleted. thanks.

**Pull request merged**